### ISSUE TITLE
Compile Error using std::thread and objects  (this)

### ISSUE THREAD

**Issue by voertler** [meganz/mingw-std-threads#37]
I can't generate a std::thread which works on an object (extra argument this). 
`std::thread t(&Hello::print, this);`
The following program runs fine under linux, however we get an error using a 4.9.3 gcc:

`mingw.thread.h:67:11: error: must use '.*' or '->*' to call pointer-to-member function in '((mingw_stdthread::detail::ThreadFuncCall<void (Hello::*)(), Hello*>*)this)->mingw_stdthread::detail::ThreadFuncCall<void (Hello::*)(), Hello*>::mFunc (...)', e.g. '(... ->* ((mingw_stdthread::detail::ThreadFuncCall<void (Hello::*)(), Hello*>*)this)->mingw_stdthread::detail::ThreadFuncCall<void (Hello::*)(), Hello*>::mFunc) (...)'
           mFunc(std::get<S>(std::forward<Tuple>(mArgs)) ...);`

```c++
#include <iostream>
#include <thread>
#include <mingw.thread.h>
struct Hello {
	int val;
	Hello(int val_): val(val_){
		std::thread t(&Hello::print, this);
		t.join();
	};
	void print () {
		std::cout << "Hello " << val << std::endl;
	}

};
int main(int argc, char **argv) {
	Hello h(1);
}
```

**Comment by nmcclatchey** [comment]
This is related to #34. Specifically, `thread` and `call_once` do not behave as if using `std::invoke`.

If you want to write a patch for it, I have the following recommendation.
- When compiling for C++17 (`#if __cplusplus >= 201703L`), use `std::invoke`.
- When compiling for C++14 and lower, a custom implementation equivalent to `std::invoke` would be required. The type_traits header allows detection of the relevant types, while a helper class will allow template specialization.

If you'd rather wait for someone else to fix it, I'll get around to it when I have time.

**Comment by nmcclatchey** [comment]
Fixed in #38.
